#### Problem 1 : Cold Storage started its operations in Jan 2016. They are in the business of storing Pasteurized Fresh, Whole or Skimmed Milk, Sweet Cream, Flavoured Milk Drinks. To ensure that there is no change of texture, body appearance, separation of fats the optimal temperature to be maintained is between 2º - 4º C. In the first year of business, they outsourced the plant maintenance work to a professional company with stiff penalty clauses. It was agreed that if it was statistically proven that the probability of temperature going outside the 2º - 4º C during the one-year contract was above 2.5% and less than 5% then the penalty would be 10% of AMC (annual maintenance case). In case it exceeded 5% then the penalty would be 25% of the AMC fee. The average temperature data at the date level is given in the file “Cold_Storage_Temp_Data_.csv”

1. Find mean cold storage temperature for Summer, Winter, and Rainy Season. (7 marks)
2. Find the overall mean for the full year. (7 marks)
3. Find Standard Deviation for the full year. (7 marks)
4. Assume Normal distribution, what is the probability of temperature having fallen below 2º C? (7 marks)
5. Assume Normal distribution, what is the probability of temperature having gone above 4º C? (7 marks)
6. What will be the penalty for the AMC Company? (5 marks)


In [1]:
# Importing librabries

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import scipy.stats as stats
from scipy.stats import ttest_1samp,ttest_ind

In [2]:
# Loading dataset

cold_storage = pd.read_csv('D:\\Study\\Data Science and Machine Learning\\Great_learning course\\Main content\\Module2-  SMDM- Statistical Methods For Decision Making\\Week 4- Project\\Cold_Storage_Temp_Data_-1 (1).csv')

In [3]:
# Checking first 5 rows of the data set.

cold_storage.head()

Season Month  Date  Temperature
0  Winter   Jan     1          2.3
1  Winter   Jan     2          2.2
2  Winter   Jan     3          2.4
3  Winter   Jan     4          2.8
4  Winter   Jan     5          2.5

In [4]:
# Checking data summary

cold_storage.describe()

Date  Temperature
count  365.000000   365.000000
mean    15.720548     3.002466
std      8.808321     0.465832
min      1.000000     1.700000
25%      8.000000     2.700000
50%     16.000000     3.000000
75%     23.000000     3.300000
max     31.000000     4.500000

In [5]:
#Checking information.

cold_storage.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 365 entries, 0 to 364
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Season       365 non-null    object 
 1   Month        365 non-null    object 
 2   Date         365 non-null    int64  
 3   Temperature  365 non-null    float64
dtypes: float64(1), int64(1), object(2)
memory usage: 11.5+ KB


In [6]:
# Checking missing values in the data set.
cold_storage.isnull().sum()

Season         0
Month          0
Date           0
Temperature    0
dtype: int64

In [7]:
# Checking duplicate values in the data set.

cold_storage.duplicated().sum()

0

In [8]:
# Converting column 'Season' into dtype 'Category'.

cold_storage.Season = pd.Categorical(cold_storage.Season)
cold_storage.Season.value_counts()

Winter    123
Rainy     122
Summer    120
Name: Season, dtype: int64

In [9]:
# Converting column 'Month into dtype 'Category'.

cold_storage.Month = pd.Categorical(cold_storage.Month)
cold_storage.Month.value_counts()

Aug    31
Dec    31
Jan    31
Jul    31
Mar    31
May    31
Oct    31
Apr    30
Jun    30
Nov    30
Sep    30
Feb    28
Name: Month, dtype: int64

#### Ques-1) Find mean cold storage temperature for Summer, Winter, and Rainy Season.

In [10]:
Mean_temp_season = cold_storage.groupby('Season')['Temperature']

In [11]:
Mean_temp_season.mean()

Season
Rainy     3.087705
Summer    3.147500
Winter    2.776423
Name: Temperature, dtype: float64

In [12]:
Mean_temp_season.describe()

count      mean       std  min   25%   50%  75%  max
Season                                                      
Rainy   122.0  3.087705  0.527207  1.7  2.70  3.05  3.4  4.5
Summer  120.0  3.147500  0.351698  2.5  2.90  3.20  3.4  4.0
Winter  123.0  2.776423  0.414310  1.8  2.45  2.80  3.0  3.9

##### Inference: 

* From the above table we can make following observations:

* The average cold storage temperature in summer season is highest that is 3.14.

* Whereas average cold storage temperature in winter season seems lowest that is 2.77.

#### Ques-2). Find the overall mean for the full year. 


In [13]:
cold_storage.describe()

Date  Temperature
count  365.000000   365.000000
mean    15.720548     3.002466
std      8.808321     0.465832
min      1.000000     1.700000
25%      8.000000     2.700000
50%     16.000000     3.000000
75%     23.000000     3.300000
max     31.000000     4.500000

In [14]:
cold_storage.Temperature.mean()

3.0024657534246546

#### Inference: 

* From the above table, it is evident that the overall mean temperature for the full year is 3.002.

#### Ques-3) Find Standard Deviation for the full year.

In [28]:
cold_storage.describe()

Date  Temperature
count  365.000000   365.000000
mean    15.720548     3.002466
std      8.808321     0.465832
min      1.000000     1.700000
25%      8.000000     2.700000
50%     16.000000     3.000000
75%     23.000000     3.300000
max     31.000000     4.500000

In [15]:
cold_storage.Temperature.std()

0.4658319416510761

Inference:

* From the above table, we can observe that the standard deviation in temperature for the full year is 0.46.

#### Ques-4) Assume Normal distribution, what is the probability of temperature having fallen below 2º C? 

In [16]:
cold_storage[cold_storage.Temperature<2].count()

Season         3
Month          3
Date           3
Temperature    3
dtype: int64

In [17]:
# Counting probability of temperature having fallen below 2 degree celcius.

(3/365)*100

0.821917808219178

#### Inference: 

* From the above calculation we can say that the probability of temperature having fallen below 2 degree celcius is only 0.8%.

#### Ques-5) Assume Normal distribution, what is the probability of temperature having gone above 4º C? 



In [18]:
cold_storage[cold_storage.Temperature>4].count()

Season         7
Month          7
Date           7
Temperature    7
dtype: int64

In [19]:
# Counting probability of temperature having gone above 4 degree celcius.

(7/365)*100

1.9178082191780823

#### Inference:

* From the above calculation, we can conclude that the probability of temperature having gone above 4 degree celcius is only 1.92%.

#### Ques-6) What will be the penalty for the AMC Company? 


#### Since, the total probability of temperature going outside of 2 degree i.e.(0.8%) and 4 degree celcius i.e.(1.91%) which on adding becomes 2.71%, therefore we can conclude that there will be 10% penalty for the AMC company because of the penalty conditions mentioned in the contract.  

#### Problem 2) In Mar 2018, Cold Storage started getting complaints from their clients that they have been getting complaints from end consumers of the dairy products going sour and often smelling. On getting these complaints, the supervisor pulls out data of the last 35 days’ temperatures. As a safety measure, the Supervisor has been vigilant to maintain the mean temperature 3.9º C or below. Assume 3.9º C as the upper acceptable mean temperature and at alpha = 0.1 do you feel that there is a need for some corrective action in the Cold Storage Plant or is it that the problem is from the procurement side from where Cold Storage is getting the Dairy Products. The data of the last 35 days is in “Cold_Storage_Mar2018_.csv”

#### 1. Which Hypothesis test shall be performed to check if corrective action is needed at the cold storage plant? Justify your answer. (6 marks)
#### 2. State the Hypothesis and do the necessary calculations to accept or reject the corresponding null hypothesis. (8 marks)
#### 3. Give your inference. (6 marks)
#### Dataset to be used for Problem 2: Cold_Storage_Mar2018_.csv

In [20]:
## Loading dataset for problem 2.

cold_storage_march = pd.read_csv('D:\\Study\\Data Science and Machine Learning\\Great_learning course\\Main content\\Module2-  SMDM- Statistical Methods For Decision Making\\Week 4- Project\\Cold_Storage_Mar2018_ (1).csv')

In [21]:
# Checking top 5 counts of the data set.

cold_storage_march.head()

Season Month  Date   Temperature 
0  Summer   Feb    11            4.0
1  Summer   Feb    12            3.9
2  Summer   Feb    13            3.9
3  Summer   Feb    14            4.0
4  Summer   Feb    15            3.8

In [22]:
cold_storage_march.columns

Index(['Season', 'Month', 'Date', ' Temperature '], dtype='object')

In [23]:
#  Checking data information.
cold_storage_march.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 35 entries, 0 to 34
Data columns (total 4 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Season         35 non-null     object 
 1   Month          35 non-null     object 
 2   Date           35 non-null     int64  
 3    Temperature   35 non-null     float64
dtypes: float64(1), int64(1), object(2)
memory usage: 1.2+ KB


In [24]:
# Checking statistical summary of the data set.

cold_storage_march.describe()

Date   Temperature 
count  35.000000      35.000000
mean   14.400000       3.974286
std     7.389181       0.159674
min     1.000000       3.800000
25%     9.500000       3.900000
50%    14.000000       3.900000
75%    19.500000       4.100000
max    28.000000       4.600000

In [25]:
cold_temp = cold_storage_march[' Temperature ']

#### 1. Which Hypothesis test shall be performed to check if corrective action is needed at the cold storage plant? Justify your answer.

#### Answer-1) To check if the corrective action is needed at the cold storage plant, we'll need to perform 'One-tailed Hypothesis test' due to the following reasons:  

* We need to perform test on one samples since we do not have the population standard deviation. 

* Sample size is greater than 30.

* We have to use the t-distribution and $t_{stat}$ test statistic for one sample. 



#### 2.  State the Hypothesis and do the necessary calculations to accept or reject the corresponding null hypothesis.

#### Step 1 : Deciding the Null and Alternate Hypothesis.

#### Null Hypothesis '$H_0$' = mean temperature of the cold storage is less than or equal to 3.9 degree celcius or 'corrective action is not needed at cold storage'.

* $H_0$ = $\mu$ $\leq$ '3.9 degree celcius'

#### Alternative Hypothesis '$H_A$' = mean temperature of the cold storage is greater than 3.9 or 'corrective action is needed at cold storage'.

* $H_A$ = $\mu$ $\gt$ '3.9 degree celcius' 

#### Step 2 : Decide the Significance Level

* Here we select, alpha '$\alpha$' = 0.01

#### Step 3 : Identify the test statistic 

* We are performing test on one sample and we do not have the population standard deviation. 

* Sample size is greater than 30.

* We have to use the t-distribution and $t_{stat}$ test statistic for one sample. 

#### Step 4 : Calculate the p_value and test statistic. 

In [26]:
## one sample t-test.

## null_hypothesis: expected mean value = 3.9

t_statistic, p_value = ttest_1samp(cold_temp,3.9)

print('one sample t-test \n"t_statistic" : {0}, "p_value" : {1}'.format(t_statistic,p_value))

one sample t-test 
"t_statistic" : 2.752358609800241, "p_value" : 0.009422395404264431


#### Ques-3) Decide to reject or accept the null hypothesis.

#### Inference-3)

In [27]:
alpha_value = 0.01 # Level of Significance
print('Level of Significance: %.2f' %alpha_value)
if p_value < alpha_value : 
    print('Since p_value is less than Level of Significance, we have enough evidence to reject the null hypothesis, therefore "corrective action is needed at cold storage".')
else:
    print('Since p_value is more than Level of Significance, we do not have enough evidence to reject the null hypothesis, therefore "corrective action is not needed at cold storage".')



Level of Significance: 0.01
Since p_value is less than Level of Significance, we have enough evidence to reject the null hypothesis, therefore "corrective action is needed at cold storage".
